hydroflow/tests/surface_scheduling.rs

```diff
@@ -121,3 +123,43 @@ pub fn test_issue_800_1050_fold_keyed() {
     assert_graphvis_snapshots!(df);
     df.run_available();
 }
+
+#[multiplatform_test(hydroflow, env_tracing)]
+async fn test_nospin_issue_961() {
+    let mut df = hydroflow_syntax! {
+        source_iter([1])
+            -> next_stratum()
+            -> persist()
+            -> defer_tick_lazy()
+            -> null();
+    };
+    assert_graphvis_snapshots!(df);
+
+    timeout(Duration::from_millis(100), df.run_available_async())
+        .await
+        .expect("Should not spin.");
```

Comment:
Is there a way to do this without measuring wall-clock time. Tests like this can become flaky over time. Perhaps something like checking over 1000s of iterations that the only tasks scheduled is the one that loops?